engine/baml-runtime/src/runtime/runtime_interface.rs

```diff
@@ -76,17 +86,34 @@ impl<'a> InternalClientLookup<'a> for InternalBamlRuntime {
                 #[cfg(not(target_arch = "wasm32"))]
                 let clients = &self.clients;
 
-                if let Some(client) = clients.get(client_name) {
-                    Ok(client.clone())
-                } else {
-                    let walker = self
-                        .ir()
-                        .find_client(client_name)
-                        .context(format!("Could not find client with name: {}", client_name))?;
-                    let client = LLMProvider::try_from((&walker, ctx)).map(Arc::new)?;
-                    clients.insert(client_name.into(), client.clone());
-                    Ok(client)
+                // if a client exists, check if the env vars have changed
+                if clients.contains_key(client_name) {
+                    // make sure to clone the client to avoid holding a lock, otherwise dashmap will deadlock!
+                    let client = clients.get(client_name).map(|c| c.clone()).unwrap();
```

Comment:
There's another thing we should `clippy` linting in CI so that it warns us about it, when you have a closure that takes in an object and simply calls a method on that object, you can avoid the closure:

```rust
// Instead of this...
let client = clients.get(client_name).map(|c| c.clone()).unwrap();

// Just pass the function
let client = clients.get(client_name).map(Clone::clone).unwrap();

// Or use StructName::clone for clarity (Clone::clone is confusing)
```

Reply:
oh this was interesting! I'll set up clippy on my machine then.
Thanks a ton!